design.md

```diff
@@ -77,7 +77,25 @@ Instead of doing a large PR that will likely be the daily chore of a single team
 #### Cons
 
 * Can run into merge conflicts galore, will probably need some system to deal with these.
-* Some tests (like complex subtests) can not be easily quarantined and my need manual intervention, which can frustrate devs who are unfamiliar with the process. This isn't a big deal, and can be ameliorated with good docs and alerts.
+* Some tests (like complex subtests) cannot be easily quarantined and may need manual intervention, which can frustrate devs who are unfamiliar with the process. This isn't a big deal, and can be ameliorated with good docs and alerts.
+
+## Guard Mode
+
+A primary goal of Flakeguard is to **guard** your CI from flakes, and guard your main branch from new flakes being introduced. When running in `guard` mode, Flakeguard will detect newly added (and modified) tests and run some `detect` loops on them. If it finds that you're introducing newly flaky tests, it will block your PR.
```

Comment:
This implies that guard also runs detects on new tests.

I think there should be a logical division between the two, guard re-runs flaky tests, detect runs the full test suite multiple times to detect flakiness.

Perhaps there can be a way to do both, but I think the two features should be separated by default?